### PR TITLE
Improve routing to boost clock rate

### PR DESCRIPTION
**Pull request by tdb-alcorn**
This change

- merges fixed point multiply and add into a single operation called mac, which avoids computing the saturation handling twice
- adds various input buffers to modules as needed to break up long routes

Ultra96v2 now makes timing at 150MHz with a WNS of 0.114ns. The longest routes now pass through the saturation handling in the acc ALUs.

Additional changes

- Re-use fixed-point MAC for times and plus;
- Golden MAC implementation to match hardware;
- Test fixed point with fuzzing;
- Fix fixed-point minus (discovered by fuzzing);
- Use ONNX ResNet-20 in simulation test;
- Use Mill fork args to allow simulation test with sufficient memory;

### CODE REVIEW

File: rtl/src/main/scala/tensil/util/package.scala [hunk 1]
```diff
@@ -147,6 +172,14 @@ package object util {
     saturated.asFixedPoint(gen.binaryPoint)
   }
 
+  def mac[T <: Data with Num[T]](gen: T, x: T, y: T, z: T): T = {
+    (gen, x, y, z) match {
+      case (gen: FixedPoint, x: FixedPoint, y: FixedPoint, z: FixedPoint) =>
+        macFixedPoint(gen, x.asSInt(), y.asSInt(), z.asSInt()).asInstanceOf[T]
+      case _ => x * y
```
Comment (petrohi) [inline rtl/src/main/scala/tensil/util/package.scala:179]: Should also add z here.

**Review by petrohi — CHANGES_REQUESTED** [review]
We also need to change golden processor accordingly to have bit accurate results. I will do it in this PR.

**Review by petrohi — APPROVED** [review]

File: sim/test/src/zynq/tcu/AXIWrapperTCUSpec.scala [hunk 2]
```diff
@@ -500,13 +500,13 @@ class AXIWrapperTCUSpec extends FunUnitSpec {
               .filter(_ <= arch.accumulatorDepth)
 
           val tests = Seq(
-            () => xor4(batchSize = 1),
+            /*() => xor4(batchSize = 1),
```
Comment (tdb-alcorn) [inline sim/test/src/zynq/tcu/AXIWrapperTCUSpec.scala:503]: uncomment these again?

Comment (petrohi) [reply sim/test/src/zynq/tcu/AXIWrapperTCUSpec.scala]: done

**Pull request merged**